### PR TITLE
Update teamcity commit status publisher settings

### PR DESCRIPTION
**Pull request by Badya**
Rotating access keys and switching to use of GitHub Apps for authentication.

### CODE REVIEW

File: .teamcity/settings.kts [hunk 1]
```diff
@@ -105,6 +105,19 @@ fun Project.buildAll(versionBuild: BuildType) = BuildType {
...
+            publisher = github {
+                githubUrl = "https://api.github.com"
+                authType = storedToken {
+                    tokenId =
+                        "tc_token_id:CID_3c2e87590eb212c2b4b597f0b6696f72:34894:ebd4e9b7-fa1d-4e09-b44a-8604fc643a7c"
+                }
+            }
+        }
+    }
+
```
Comment (dkhalanskyjb) [inline .teamcity/settings.kts:120]: It's better not to edit this file directly. It's copied from https://github.com/Kotlin/kotlinx.team.infra/tree/master/main/resources/teamcity periodically, and we'd have to merge changes manually if we don't store project-specific files in `additionalConfiguration`.

**Comment by dkhalanskyjb** [comment]
@Badya, hi, is this good to go?

File: .teamcity/additionalConfiguration.kt [hunk 2]
```diff
@@ -3,19 +3,20 @@
...
 fun Project.additionalConfiguration() {
     knownBuilds.buildAll.features {
         commitStatusPublisher {
+            vcsRootExtId = "${DslContext.settingsRoot.id}"
             publisher = github {
                 githubUrl = "https://api.github.com"
-                authType = personalToken {
-                    token = "credentialsJSON:af36802a-ccd4-401b-86b9-0b08d2dfad17"
+                authType = storedToken {
+                    tokenId = "tc_token_id:CID_3c2e87590eb212c2b4b597f0b6696f72:34894:ebd4e9b7-fa1d-4e09-b44a-8604fc643a7c"
```
Comment (ilya-g) [inline .teamcity/additionalConfiguration.kt:17]: Where one can manage these tokens?

Comment (Badya) [reply .teamcity/additionalConfiguration.kt:17]: Unfortunately, the only way to get a token is via TC UI

**Review by dkhalanskyjb — APPROVED** [review]

**Pull request merged**